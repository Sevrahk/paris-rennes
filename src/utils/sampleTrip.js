export default {"links":{"_self":"https://api.blablacar.fr/api/v2/trips/1701132340-bois-colombes-rennes","_front":"https://www.blablacar.fr/trip/carpooling/1701132340-bois-colombes-rennes/"},"user":{"links":{"_front":"https://www.blablacar.fr/user/show/73ff8c00-4496-11e7-a000-003970b5dcfa","_self":"https://api.blablacar.fr/api/v2/users/gn-tpfFez_Mdvnnfa1vxgg"},"display_name":"Olivier","gender":"_UE_M","picture":"https://d2kwny77wxvuie.cloudfront.net/user/Jcip0ZFYRuqvcXgXWaKPmw/thumbnail_144x144.jpeg","has_picture":true,"smoking":"_UE_MAYBE","music":"_UE_YES","encrypted_id":"gn-tpfFez_Mdvnnfa1vxgg","pets":"_UE_MAYBE","dialog":"_UE_MAYBE","age":50,"rating":5,"rating_count":1,"grade":1,"phone_hidden":false,"id_checked":"VALID","bbpro":false,"bbpro_segment":"not_pro","uuid":"73ff8c00-4496-11e7-a000-003970b5dcfa"},"departure_date":"14/09/2019 16:00:00","departure_place":{"city_name":"Bois-Colombes","address":"13 Allée des Passiflores, 92270 Bois-Colombes, France","latitude":48.925689697,"longitude":2.272836924,"country_code":"FR"},"arrival_place":{"city_name":"Rennes","address":"La Poterie","latitude":48.087497711,"longitude":-1.644595981,"country_code":"FR"},"price":{"value":22,"currency":"EUR","symbol":"€","string_value":"22 €","price_color":"BLACK"},"price_with_commission":{"value":27,"currency":"EUR","symbol":"€","string_value":"27,00 €","price_color":"BLACK"},"price_without_commission":{"value":22,"currency":"EUR","symbol":"€","string_value":"22 €","price_color":"BLACK"},"commission":{"value":5,"currency":"EUR","symbol":"€","string_value":"5,00 €"},"seats_left":3,"seats":3,"duration":{"value":13800,"unity":"s"},"distance":{"value":350,"unity":"km"},"permanent_id":"1701132340-bois-colombes-rennes","car":{"id":"139986481","model":"CAPTUR","make":"RENAULT","comfort":"normal","comfort_nb_star":2,"category":"CATEGORY_44"},"multimodal_id":{"source":"CARPOOLING","id":"1701132340-bois-colombes-rennes"},"viaggio_rosa":false,"is_comfort":true,"freeway":true,"answer_delay":0,"booking_mode":"auto","booking_type":"online","locations_to_display":["Bois-Colombes","Rennes"],"departure_passenger_routing":{"proximity":"MIDDLE","routes":[{"distance_in_meters":9627,"type":"AS_THE_CROW_FLIES"}]},"arrival_passenger_routing":{"proximity":"MIDDLE","routes":[{"distance_in_meters":4127,"type":"AS_THE_CROW_FLIES"}]},"corridoring_type":"PLANNED","vehicle_pictures":[]};
