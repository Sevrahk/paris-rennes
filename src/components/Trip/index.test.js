import React from 'react';
import { shallow } from 'enzyme'
import Component from './index';
import sampleTrip from '../../utils/sampleTrip';
import userFallbackImg from '../../assets/img/user.png';
import { cloneDeep } from 'lodash';

describe('Trip component', () => {
    let component;

    beforeEach(() => {
        component = shallow(<Component trip={sampleTrip} />);
    });

    test('renders', () => {
        expect(component.find('.trip')).toHaveLength(1);
        expect(component.find('img').prop('src')).not.toEqual(userFallbackImg);
    });

    test('renders without user image', () => {
        let trip = cloneDeep(sampleTrip);
        trip.user.has_picture = false;

        component.setProps({trip: trip})

        expect(component.find('.trip')).toHaveLength(1);
        expect(component.find('img').prop('src')).toEqual(userFallbackImg);
    });

    test('preserve UI', () => {
        expect(component).toMatchSnapshot();
    });
});
