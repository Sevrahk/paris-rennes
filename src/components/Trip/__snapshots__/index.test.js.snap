// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trip component preserve UI 1`] = `
<div
  className="col-12 col-sm-6"
>
  <article
    className="trip"
  >
    <img
      alt="Olivier"
      src="https://d2kwny77wxvuie.cloudfront.net/user/Jcip0ZFYRuqvcXgXWaKPmw/thumbnail_144x144.jpeg"
    />
    <br />
    Olivier
    <br />
    <Stars
      rating={5}
      ratingCount={1}
    />
    <br />
    <b>
      27,00 €
    </b>
    <div
      className="path"
    >
      <span>
        Bois-Colombes
      </span>
      <span>
        Rennes
      </span>
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              448,
              512,
              Array [],
              "f1ad",
              "M436 480h-20V24c0-13.255-10.745-24-24-24H56C42.745 0 32 10.745 32 24v456H12c-6.627 0-12 5.373-12 12v20h448v-20c0-6.627-5.373-12-12-12zM128 76c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40c0 6.627-5.373 12-12 12h-40c-6.627 0-12-5.373-12-12V76zm0 96c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40c0 6.627-5.373 12-12 12h-40c-6.627 0-12-5.373-12-12v-40zm52 148h-40c-6.627 0-12-5.373-12-12v-40c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40c0 6.627-5.373 12-12 12zm76 160h-64v-84c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v84zm64-172c0 6.627-5.373 12-12 12h-40c-6.627 0-12-5.373-12-12v-40c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40zm0-96c0 6.627-5.373 12-12 12h-40c-6.627 0-12-5.373-12-12v-40c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40zm0-96c0 6.627-5.373 12-12 12h-40c-6.627 0-12-5.373-12-12V76c0-6.627 5.373-12 12-12h40c6.627 0 12 5.373 12 12v40z",
            ],
            "iconName": "building",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              448,
              512,
              Array [],
              "f30b",
              "M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z",
            ],
            "iconName": "long-arrow-alt-right",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              640,
              512,
              Array [],
              "f5e4",
              "M544 192h-16L419.22 56.02A64.025 64.025 0 0 0 369.24 32H155.33c-26.17 0-49.7 15.93-59.42 40.23L48 194.26C20.44 201.4 0 226.21 0 256v112c0 8.84 7.16 16 16 16h48c0 53.02 42.98 96 96 96s96-42.98 96-96h128c0 53.02 42.98 96 96 96s96-42.98 96-96h48c8.84 0 16-7.16 16-16v-80c0-53.02-42.98-96-96-96zM160 432c-26.47 0-48-21.53-48-48s21.53-48 48-48 48 21.53 48 48-21.53 48-48 48zm72-240H116.93l38.4-96H232v96zm48 0V96h89.24l76.8 96H280zm200 240c-26.47 0-48-21.53-48-48s21.53-48 48-48 48 21.53 48 48-21.53 48-48 48z",
            ],
            "iconName": "car-side",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              448,
              512,
              Array [],
              "f30b",
              "M313.941 216H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h301.941v46.059c0 21.382 25.851 32.09 40.971 16.971l86.059-86.059c9.373-9.373 9.373-24.569 0-33.941l-86.059-86.059c-15.119-15.119-40.971-4.411-40.971 16.971V216z",
            ],
            "iconName": "long-arrow-alt-right",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              640,
              512,
              Array [],
              "f64f",
              "M616 192H480V24c0-13.26-10.74-24-24-24H312c-13.26 0-24 10.74-24 24v72h-64V16c0-8.84-7.16-16-16-16h-16c-8.84 0-16 7.16-16 16v80h-64V16c0-8.84-7.16-16-16-16H80c-8.84 0-16 7.16-16 16v80H24c-13.26 0-24 10.74-24 24v360c0 17.67 14.33 32 32 32h576c17.67 0 32-14.33 32-32V216c0-13.26-10.75-24-24-24zM128 404c0 6.63-5.37 12-12 12H76c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12H76c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12H76c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm128 192c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm160 96c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12V76c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm160 288c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40zm0-96c0 6.63-5.37 12-12 12h-40c-6.63 0-12-5.37-12-12v-40c0-6.63 5.37-12 12-12h40c6.63 0 12 5.37 12 12v40z",
            ],
            "iconName": "city",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </div>
  </article>
</div>
`;
